webapp/channels/src/components/channel_bookmarks/bookmark_icon.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import styled from 'styled-components';
+
+import {FileGenericOutlineIcon, BookOutlineIcon} from '@mattermost/compass-icons/components';
+import type {ChannelBookmark} from '@mattermost/types/channel_bookmarks';
+import type {FileInfo} from '@mattermost/types/files';
+
+import RenderEmoji from 'components/emoji/render_emoji';
+import FileThumbnail from 'components/file_attachment/file_thumbnail';
+import type {FilePreviewInfo} from 'components/file_preview/file_preview';
+
+type Props = {
+    type: ChannelBookmark['type'];
+    emoji?: string;
+    imageUrl?: string;
+    fileInfo?: FileInfo | FilePreviewInfo;
+    size?: 16 | 24;
+}
+
+const BookmarkIcon = ({
+    type,
+    emoji,
+    imageUrl,
+    fileInfo,
+    size = 16,
+}: Props) => {
+    let icon = type === 'link' ? <BookOutlineIcon size={size}/> : <FileGenericOutlineIcon size={size}/>;
+
+    const emojiName = emoji && emoji.startsWith(':') && emoji.endsWith(':') ? emoji?.slice(1, -1) : emoji;
+
+    if (emojiName) {
+        icon = (
+            <RenderEmoji
+                emojiName={emojiName}
+                size={size}
+            />
+        );
+    } else if (imageUrl) {
+        icon = (
+            <BookmarkIconImg
+                src={imageUrl}
```

Comment:
what if this fails? should we show the bookmark outline icon when that happens ?